[Locale]Available locales in channel by api (#12766)
Sylius/Sylius: adr/2021_07_05_api_providing_locales_available_in_active_channel.md

# Using data provider for getting available locales in active channel for shop user

* Status: proposed
* Date: 2021-07-05

## Context and Problem Statement
Customer should have access only to locales available in their channel

## Considered Options

### Using Doctrine Collection extension

> * Good, because consistent with actual approach to modifying responses content

Reviewer: ```suggestion
* Good, because it is consistent with the actual approach to modifying responses content
```

> * Good, because works with rest of API extension like pagination

Reviewer: ```suggestion
* Good, because it works with the rest of API extensions like pagination
```

* Bad, because locales don't have relation to channel, so using Doctrine Collection extension is extremely hard.

### Using Data Provider

* Good, because we already have this approach for older resources

> * Good, because easy to implement 

Reviewer: ```suggestion
* Good, because it is easy to implement 
```
